homeassistant/components/ollama_conversation/config_flow.py

```diff
@@ -0,0 +1,155 @@
+
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_URL): TextSelector(
+            TextSelectorConfig(type=TextSelectorType.URL)
+        ),
+        vol.Required(CONF_MODEL): cv.string,
+        vol.Required(
+            CONF_PROMPT, description={"suggested_value": DEFAULT_PROMPT}
+        ): TemplateSelector(),
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> None:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    client = ollama.AsyncClient(host=data[CONF_URL])
+    async with asyncio.timeout(DEFAULT_TIMEOUT):
+        await client.list()
+
+
+class OllamaConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Ollama conversation."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            await validate_input(self.hass, user_input)
+        except (TimeoutError, httpx.ConnectError):
+            errors["base"] = "cannot_connect"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            return self.async_create_entry(title="Ollama Conversation", data=user_input)
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
+
+    @staticmethod
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> OptionsFlow:
+        """Create the options flow."""
+        return OllamaOptionsFlow(config_entry)
+
+
+class OllamaOptionsFlow(OptionsFlow):
+    """Ollama conversation options flow."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(title="Ollama Conversation", data=user_input)
+
+        options = self.config_entry.options or self.config_entry.data
+        schema = ollama_config_option_schema(MappingProxyType(options))
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(schema),
+        )
+
+
+def ollama_config_option_schema(options: MappingProxyType[str, Any]) -> dict:
+    """Ollama conversation options schema."""
+    return {
+        vol.Required(
+            CONF_URL,
+            description={"suggested_value": options[CONF_URL]},
```

Comment:
Should we allow changing URL? That is not an option but an authentication thing? This should be a reconfigure flow and out of scope for this PR.